finality-aleph/src/justification/compatibility.rs

```diff
@@ -34,9 +34,11 @@ pub enum JustificationDecoding {
 }
 
 pub fn backwards_compatible_decode(justification_raw: Vec<u8>) -> JustificationDecoding {
-    if let Ok(justification) = AlephJustification::decode_all(&justification_raw) {
+    if let Ok(justification) = AlephJustification::decode_all(&mut justification_raw.as_slice()) {
```

Comment:
The `mut` here is somewhat worrisome. No idea how decoding might mutate this, but if it actually does then it's hard to tell what the second decode below might do. Perhaps cloning would be prudent.